tollway/__main__.py

```diff
@@ -0,0 +1,124 @@
+import time
+import random
+import json
+from typing import Optional
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import typer
+from faker import Faker
+from faker_vehicle import VehicleProvider
+
+from tollway.vehicle import (
+    get_tollways, 
+    create_tollway, 
+    create_vehicle, 
+    create_payload,
+    TIMESTAMP_FORMAT,
+)
+
+
+DATE_VARIATION_RATE = 3
+INCLUDE_LATE_RATE = 20
+INCLUDE_DUPLICATE_RATE = 50
+ALL_EVENTS_COUNT = 100
+
+tollways = get_tollways()
+fake = Faker()
+fake.add_provider(VehicleProvider)
+
+
+def _get_random_timestamp(timestamps: list[str]) -> str:
+    first_timestamp = datetime.strptime(timestamps[0], TIMESTAMP_FORMAT)
+    middle_timestamp = datetime.strptime(timestamps[len(timestamps) // 2], TIMESTAMP_FORMAT)
+
+    random_percentage = random.random()
+    timestamp_range = middle_timestamp - first_timestamp
+    random_timestamp = (first_timestamp + random_percentage * timestamp_range).strftime(TIMESTAMP_FORMAT)
+    return random_timestamp
+
+
+def _push_to_topic():
+    pass # to be developed
+
+
+def _write_to_file(filename: str, events: list[dict]):
+    path = str(Path().absolute())
+    with open(f"{path}/{filename}", mode="a") as file:
+        json.dump(obj=events, fp=file, indent=4)
+
+
+def _date_variation(timestamp: str) -> str:
+    random_integer = random.randint(1, 3)
+    current_timestamp = datetime.strptime(timestamp, TIMESTAMP_FORMAT)
+    updated_timestamp = (current_timestamp - timedelta(days=random_integer)).strftime(TIMESTAMP_FORMAT)
+    return updated_timestamp
+
+
+def main(
+        total_events: Optional[int] = typer.Option(default=1),
+        event_rate: Optional[float] = typer.Option(default=1.0),
+        output_file: Optional[bool] = typer.Option(default=False),
+        output_filename: Optional[str] = typer.Option(default="tollway-traffic.json"),
+        date_variation: Optional[bool] = typer.Option(default=False),
+        include_late: Optional[bool] = typer.Option(default=False),
+        include_duplicate: Optional[bool] = typer.Option(default=False),
+        ):
+
+    tollway = create_tollway(tollways)
+    date_variation_events = []
+    past_events_timestamps = []
+    past_events = []
+    all_events = []
+
+    for event_count in range(total_events):
```

Comment:
this is quite long (high cyclomatic complexity), can you refactor this to simplify?

Reply:
on the other hand this is main / cli logic, so it can stay this way for now